nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -0,0 +1,176 @@
+
+
+class FindSitesWithMaintenanceOverlap(Job):
+    """Nautobot Job definition for finding sites without redundant circuit for impactful maintenance.
+
+    Args:
+        Job (Nautobot Job): Nautobot Job parent class
+    """
+
+    class Meta:
+        """Meta definition for the Job."""
+
+        name = "Find Sites With Circuit Overlap"
+        description = "Search for sites with multiple circuits, 1 or more circuit impacts."
+
+    def run(self, data=None, commit=None):
+        """Executes the Job.
+
+        Args:
+            data (_type_, optional): _description_. Defaults to None.
+            commit (_type_, optional): _description_. Defaults to None.
+        """
+        # Query for all of the circuits maintenances that are on going in the future
+        today = date.today()
+        circuit_maintenances = CircuitMaintenance.objects.filter(start_time__gte=today).order_by("start_time")
+
+        # Build a circuit mapper
+        circuit_maintenance_mapper = build_sites_to_maintenance_mapper(circuit_maintenances)
+
+        # Loop over each of the circuit maintenance records
+        # pylint: disable=too-many-nested-blocks
+        counter = 0
+        for circuit_maint in circuit_maintenances:
+            counter += 1
+            # Get the list of sites
+            sites: set = set()
+            for circuit in circuit_maint.circuits:
+                sites.add(get_sites_from_circuit(circuit))
+
+            # Check to see if there are any circuit maintenances that are overlapping at some moment in time
+            for site in sites:
+                for other_circuit_maint in circuit_maintenance_mapper[site.name]:
+                    # Report failures for any time where a circuit will take an outage
+                    if circuit_maint == other_circuit_maint:
+                        continue
+                    if check_for_overlap(circuit_maint, other_circuit_maint):
+                        if PLUGIN_SETTINGS.get("overlap_job_log_only_impacts"):
+                            for impact in circuit_maint.circuitimpact_set:
+                                if impact.impact in CIRCUIT_IMPACT_SET:
+                                    self.log_warning(
+                                        obj=site,
+                                        message=f"There is an overlapping maintenance for site: {site.name}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
+                                    )
+                        else:
+                            self.log_warning(
+                                obj=site,
+                                message=f"There is an overlapping maintenance for site: {site.name}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
+                            )
+                    else:
+                        # Log success for each time there is a known circuit still available at the site at the same time
+                        self.log_info(
```

Comment:
I still think this should be "debug", so you could easily exclude with the `debug` checkbox